src/core/core.controller.js

```diff
@@ -933,7 +973,26 @@ module.exports = function(Chart) {
 			me.lastActive = me.active;
 
 			return changed;
-		}
+		},
+		/**
+		 * Format the data list if the data is missing some points in the datasets.
+		 * @private
+		 * @param {array} dataArray array to format
+		 * @return {array} the formated array
+		 */
+		_formatDataset: function(dataArray) {
+			var labels = this.chart.data.labels;
+			var tmp = dataArray.slice(0);
```

Comment:
there's no reason to create `tmp` here as a copy of `dataArray`. the only actions taken on it are read-only. just use `dataArray` directly and remove `tmp`